iocore/net/SSLUtils.cc

```diff
@@ -2223,6 +2223,7 @@ SSLWriteBuffer(SSL *ssl, const void *buf, int64_t nbytes, int64_t &nwritten)
     }
     return SSL_ERROR_NONE;
   }
+  nwritten      = -errno;
   int ssl_error = SSL_get_error(ssl, ret);
   if (ssl_error == SSL_ERROR_SSL && is_debug_tag_set("ssl.error.write")) {
```

Comment:
There are `lock()` and `unlock()` within `diags->on()`. The macro `is_debug_tag_set` is defined to it if `TS_USE_DIAGS` defined.

The `lock()` and `unlock()` is wrapper for `ink_mutex_acquire()` and `ink_mutex_release()`.

Reply:
if ```SSL_ERROR_SSL``` is returned, ```errno``` maybe overwrote by ```ink_mutex_acquire()``` and ```ink_mutex_release()```.